cdap-cli/src/main/java/co/cask/cdap/cli/command/HelpCommand.java

```diff
@@ -67,4 +73,93 @@ public String getPattern() {
   public String getDescription() {
     return "Prints this helper text";
   }
+
+  /**
+   * Prints the given command pattern which wrap at the given column width. It prints on multiple lines like this:
+   *
+   * <pre>{@code
+   * command sub-command <arg1> <arg2>
+   *                     <arg3> [<optiona-long-arg4>]
+   * }
+   * </pre>
+   *
+   * @param pattern the command pattern to print
+   * @param output the {@link PrintStream} to write to
+   * @param colWidth width of the column
+   */
+  private void printPattern(String pattern, PrintStream output, int colWidth) {
+    if (pattern.length() <= colWidth) {
+      output.println(pattern);
+      return;
+    }
+
+    // Find the first <argument>, it is used for alignment in 2nd line and onward.
+    int startIdx = pattern.indexOf('<');
+    // If no '<', it shouldn't reach here (they should be short command). If it does, just print it.
+    if (startIdx < 0) {
+      output.println(pattern);
+      return;
+    }
+
+    // First line should at least includes the first <argument>
+    int idx = pattern.lastIndexOf('>', colWidth) + 1;
+    if (idx <= 0) {
+      idx = pattern.indexOf('>', startIdx) + 1;
+      if (idx <= 0) {
+        idx = pattern.length();
+      }
+    }
+    // Make sure we includes the closing ] of optional argument
+    if (idx < pattern.length() && pattern.charAt(idx) == ']') {
+      idx++;
+    }
+    output.println(pattern.substring(0, idx));
+
+    if ((idx + 1) < pattern.length()) {
+      // For rest of the line, align them to the startIdx
+      wrappedPrint(pattern.substring(idx + 1), output, colWidth, startIdx);
+    }
+  }
+
+  /**
+   * Prints the given string with auto line wrapping at the given column width.
+   *
+   * @param str the string to print
+   * @param output the {@link PrintStream} to write to
+   * @param colWidth width of the column
+   * @param prefixSpaces number of spaces as the prefix for each line printed
+   */
+  private void wrappedPrint(String str, PrintStream output, int colWidth, int prefixSpaces) {
+    String prefix = Strings.repeat(" ", prefixSpaces);
+    colWidth -= prefixSpaces;
+
+    if (str.length() <= colWidth) {
+      output.printf("%s%s", prefix, str);
+      output.println();
+      return;
+    }
+
+    int beginIdx = 0;
+    while (beginIdx < str.length()) {
+      int idx;
+      if (beginIdx + colWidth >= str.length()) {
+        idx = str.length();
+      } else {
+        idx = str.lastIndexOf(' ', beginIdx + colWidth);
+      }
+
+      // Cannot break line if no space found between beginIdx and (beginIdx + colWidth)
+      // The best we can do is to look forward.
+      // The line will be longer than colWidth though.
```

Comment:
it will break the word. The best is to insert hyphen, however, I don't think we have word longer than 80 characters, no until we ported to support German.